compiler/fir/fir2ir/src/org/jetbrains/kotlin/fir/backend/Fir2IrTypeConverter.kt

```diff
@@ -99,7 +99,7 @@ class Fir2IrTypeConverter(
                 intersectedTypes.first().toIrType(typeContext)
             }
             is ConeStubType -> createErrorType()
-            is ConeIntegerLiteralType -> createErrorType()
+            is ConeIntegerLiteralType -> getApproximatedType().toIrType(typeContext)
```

Comment:
Normally we should not have ILT at this stage; if we do, then it's (probably) a front-end resolve issue // cc @demiurg906

Reply:
Thank you for pointing this out. (Actually I expected you will do so, since I found this one is intentionally generating error type previously.)

This is indeed related to inverted array type and element type for `IrVararg` and my change about computing the common super type of all elements as the array's element type. If I revert this, Ir text generation test with `arrayInAnnotationArguments.kt` fails, complaining that part of `commonSupertype` is error type. In that test, we can see the simple example: `@TestAnnWithIntArray([4, 5, 6])`.

So, even without this change, for `[4, 5, 6]`, we're seeing `FirArrayOfCall` with integer literal arguments. Creating an error type (from retrieving the first element and converting its type) hid the resolution issue, if ILT should not be at this stage.

Nonetheless, I still believe seeing the first element only should be an issue, and the element type should be the least upper bound of all arguments in the array.

Reply:
Actually, returning a non-error type for ILT would be rather hiding an issue. Since I described the issue above, i.e., an example with the concrete failing test, I can revert this part, and leave some TODOs in element type computation. Updated this commit accordingly.